pkg/hope/hypervisors/esxi_hypervisor.go

```diff
@@ -18,11 +19,32 @@ import (
 
 type EsxiHypervisor struct {
 	node hope.Node
+
+	insecure bool
 }
 
 func (hyp *EsxiHypervisor) Initialize(node hope.Node) error {
 	hyp.node = node
-	return nil
+
+	errs := []error{}
+	pm := ParameterMap(node.Parameters)
+	if insecure, ok := pm["INSECURE"]; ok {
+		switch insecure {
+		case "true", "1":
+			hyp.insecure = true
+		case "false", "0":
+			hyp.insecure = false
+		default:
+			errs = append(errs, fmt.Errorf("unknown value '%s' for INSECURE in ESXI hypervisor", insecure))
+		}
+		delete(pm, "INSECURE")
+	}
+
+	for key := range pm {
```

Comment:
Could push this into `ParameterMap` if I give it a list of valid property names. Save every hypervisor from needing to implement the same.